### ISSUE TITLE
TypeError when formatting numbers <= 1e-7

### ISSUE THREAD

**Issue by timbowhite** [spalger/numeral-js#6]
Hi thanks for forking and developing numeral-js. Using this module b/c the original numeral-js has been abandoned by its maintainer.

Just ran into this:
```
const numeral = require('@elastic/numeral');
console.log(numeral('0.0000001').format('0,0.00'));
```
Error:
```
node_modules/@elastic/numeral/numeral.js:378
                if (d.split('.')[1].length) {
                                   ^

TypeError: Cannot read property 'length' of undefined
    at formatNumber (node_modules/@elastic/numeral/numeral.js:378:36)
...
```

Same thing happens when formatting ```1e-7``` or smaller.

**Comment by spalger** [comment]
Ooo, thanks for tracking this down. @w33ble have time to take a look?

**Comment by w33ble** [comment]
Not today. I might try to get to it some time next week. 

@timbowhite do you happen to know if this is a problem with the upstream numeral as well?

**Comment by timbowhite** [comment]
@w33ble 
Seems to be. Upstream numeral formats `1e-7`, `'0.0000001'`, and smaller as `NaN`:

[Issue #512](https://github.com/adamwdraper/Numeral-js/issues/512)

A few PR's:
[#514](https://github.com/adamwdraper/Numeral-js/pull/514)
[#564](https://github.com/adamwdraper/Numeral-js/pull/564)

I'll have some time later this week to dig in if you're unable to.

**Comment by w33ble** [comment]
If you want to take a stab at it, feel free. We should probably come to a decision about what should happen though. 

Looking at that issue and the PRs there, I don't think any of those is what we'd want to see happen. `1e-7` is a valid number, so return `NaN` seems wrong. This seems really similar to #3, just with really small numbers instead of really large numbers. I'd probably fix it the same way as #4, and just return the value as a string, untouched.

@spalger any thoughts on this?

**Comment by spalger** [comment]
> I'd probably fix it the same way as #4, and just return the value as a string, untouched.

I like that much more than returning `NaN`

**Comment by tsullivan** [comment]
I need a workaround for this, and I'm thinking I'll just have to process the number before giving it to numeral with something like
```
num = Math.max(num, 1e-6);
```
